### DISCUSSION TITLE
daps multi languages support

### DISCUSSION THREAD

**Discussion by dnkbln** [openSUSE/daps#623] [Q&A] [answered]
Hello,
I start a documentation project and I want support two or more languages. I saw in the daps documentation the locdrop command. But I don't know which DEF file to specify. I have only a DC_File with my project definition. I use the daps-init command to initialize my file structure and the I wrote the XML files. 
What must be defined in the DEF File or should I use something else?
Thanks,
Dirk

**Comment by tomschr** [comment]
@dnkbln Hi Dirk and sorry for the delay.

Yes, we have this locdrop command which is an abbreviation for _localization drop_.

> I don't know which DEF file to specify.

It's a very specific to our workflow at SUSE. I doubt it would be useful to you. :wink: 

But just for the sake of completeness, a DEF file looks like this (taken from the `SUSE/doc-sle repo/DEF-opensuse-manuals`):

```
html        opensuse-all              common_copyright_gfdl.xml ok https://api.suse.de openSUSE:Factory
pdfsub      opensuse-autoyast         common_copyright_opensuse.xml
pdfsub      opensuse-gnomeuser        common_copyright_opensuse.xml
pdfsub      opensuse-refrence         common_copyright_opensuse.xml
pdfsub      opensuse-security         common_copyright_opensuse.xml
pdfsub      opensuse-startup          common_copyright_opensuse.xml
pdfsub      opensuse-tuning           common_copyright_opensuse.xml
pdfsub      opensuse-virtualization   common_copyright_opensuse.xml
```

These days you only need the first two columns:

1. column: type of deliverable; `pdfsub` causes the packaging script to create a subpackage with the PDF of that book
2. the DC file (without the `DC-` prefix)
3. the XML file copyright info
4. ???
5. The build server instance (internal/external)
6. project name in OBS


> What must be defined in the DEF File or should I use something else?

Have you tried the `package-src` subcommand?

```shell
$ daps -v -d DC-FILE package-src
```

This command creates a tar.bz2 file with all images, XML files, and the DC file. Not sure if that is the solution you are looking for.

Sorry again for the delay. Hope it will help to solve your questions.

**Comment by tomschr** [comment]
If you have further questions, you can also use our newly create discussion forum:
https://github.com/openSUSE/daps/discussions

**Comment by fsundermeyer** [comment, marked as answer]
Sorry, for replying so late, I overlooked your question ;-(.

You only need the locdroip command and the DEF files if you have a set of books and articles that only gets partially translated. for this usecase the locdrop command creates separate tarballs for files and images that get translated and for those that do not get translated. Thus you always have the full set (needed to build all books) while at the same time have seperated translated and untranslated files.
The locdrop command needs to be run for each book/article in the set that gets translated. With that DAPS can only include the DC file that was specified on the command line, DC files for the other books/articles in the set are not included. Optionally specifying the DEF file adds all DC files for the set. We are using these DEF files for packaging and they contain all DC files from the set.

Having said that, locdrop was written for a very special usecase. If you only have a single book or article for translation or if you translate a complete set, it is not needed. In both cases you can just use the DAPS command package-src to package all files. 
If you have a set and want to include all DC files, create a DEF file with the following format...

```
1 foo
2 bar
3 foobar
```

... providing you DC-files are named DC-foo, DC-bar, DC-foobar and name it `DEF-<something>` (replace `<something>`) and run 

`daps -d DC-foo package-src --def-file DEF-<something>`

Note that the DEF file may contain comment lines starting with `#`

HTH

Note: toms answered simultaneously. He is describing the DEF-file format we are using for packaging. However, it does not need to be so complicated. The format I described above also works. I am using awk to read the second column and prepend "DC-" (`awk '/^[ \t]*#/ {next};NF {printf "DC-%s ", $2}'`to be precise)

**Comment by dnkbln** [comment]
Hello,

thank you for your answer.
I will use 
`$ daps -v -d DC-FILE package-src`
to create a tar-File and use this for the translation.
